internal/controller/pdbwatcher_controller.go

```diff
@@ -161,7 +170,27 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	// What will return us to the original state? Assuming we need a watch on PDB for it to come off
+	// Watch for changes in PDB to revert to original state
+	if pdb.Status.DisruptionsAllowed > 0 && *deployment.Spec.Replicas != pdbWatcher.Status.MinReplicas {
```

Comment:
Technically a user could change replicas even when distruptions allowed == 0. In fact this might be a reasonable thing for them to do (oh I'm in trouble and my pods are failign lets through soe more in.).  So mybe you just need minreplicas != replicas (and maybe minreplicas would make more sense as original replicas?)

Reply:
If have different thoughts around disruptions allowe then comment as to why.